From stable-bounces@linux.kernel.org Fri Jul 28 15:08:08 2006
Message-ID: <44CA8A59.40002@intel.com>
Date: Fri, 28 Jul 2006 15:06:17 -0700
From: Auke Kok <auke-jan.h.kok@intel.com>
To: Greg KH <greg@kroah.com>, stable@kernel.org
Cc: Auke Kok <auke-jan.h.kok@intel.com>, NetDev <netdev@vger.kernel.org>,
        "John W. Linville" <linville@tuxdriver.com>,
        Jesse Brandeburg <jesse.brandeburg@intel.com>,
        "Ronciak,
	John" <john.ronciak@intel.com>,
        "Kok, Auke" <auke@foo-projects.org>
Subject: e1000: add forgotten PCI ID for supported device

From: Auke Kok <auke-jan.h.kok@intel.com>

The Intel(R) PRO/1000 82572EI card is fully supported by 7.0.33-k2 and
onward.  Add the device ID so this card works with 2.6.17.y onward. This
device ID was accidentally omitted.

Signed-off-by: Auke Kok <auke-jan.h.kok@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/net/e1000/e1000_hw.c |    1 +
 drivers/net/e1000/e1000_hw.h |    1 +
 2 files changed, 2 insertions(+)

--- linux-2.6.17.7.orig/drivers/net/e1000/e1000_hw.c
+++ linux-2.6.17.7/drivers/net/e1000/e1000_hw.c
@@ -353,6 +353,7 @@ e1000_set_mac_type(struct e1000_hw *hw)
     case E1000_DEV_ID_82572EI_COPPER:
     case E1000_DEV_ID_82572EI_FIBER:
     case E1000_DEV_ID_82572EI_SERDES:
+    case E1000_DEV_ID_82572EI:
         hw->mac_type = e1000_82572;
         break;
     case E1000_DEV_ID_82573E:
--- linux-2.6.17.7.orig/drivers/net/e1000/e1000_hw.h
+++ linux-2.6.17.7/drivers/net/e1000/e1000_hw.h
@@ -462,6 +462,7 @@ int32_t e1000_check_phy_reset_block(stru
 #define E1000_DEV_ID_82572EI_COPPER      0x107D
 #define E1000_DEV_ID_82572EI_FIBER       0x107E
 #define E1000_DEV_ID_82572EI_SERDES      0x107F
+#define E1000_DEV_ID_82572EI             0x10B9
 #define E1000_DEV_ID_82573E              0x108B
 #define E1000_DEV_ID_82573E_IAMT         0x108C
 #define E1000_DEV_ID_82573L              0x109A
